composer.json

```diff
@@ -11,16 +11,16 @@
         }
     ],
     "require": {
-        "php": "^7.1",
-        "guzzlehttp/psr7": "^1.4",
-        "php-http/client-common": "^2.0",
-        "psr/log": "^1.0",
-        "symfony/filesystem": "^3.4|^4.0",
-        "symfony/options-resolver": "^3.4|^4.0"
+        "php": "^7.0",
+        "php-http/client-common": "^1.9|^2.0",
+        "guzzlehttp/psr7": "^1.5",
+        "psr/log": "^1.1",
+        "symfony/filesystem": "^2.8|^3.4|^4.0",
```

Comment:
IMO adding support for Symfony versions that are not supported anymore doesn't make much sense.

Reply:
okay, but what about projects with 2.8 and currently not planned (or never planned) upgraded to 3.*?

Reply:
main point is have a branch or tag with work functional for all versions. or use separate tag or branch (for example v1 or v2) for "new" components or modern versions...

Reply:
IMO if one decides to stay on unsupported framework versions, one also needs to accept that they cannot use new features.

Reply:
okay